bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,323 @@
+                        updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                                muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+                        updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                                muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+                    } else {
+                        if (isAudioMute) {
+                            logger.trace("Received audio mute command" + command);
+                            boolean muteOn = command == OnOffType.ON;
+                            device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+                        }
+                        if (isVideoMute) {
+                            logger.trace("Received video mute command" + command);
+                            boolean muteOn = command == OnOffType.ON;
+                            device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+                        }
+                    }
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ResponseException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ConfigurationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        } catch (AuthenticationException e) {
+            this.handleAuthenticationException(e);
+        }
+    }
+
+
+    @Override
+    public void initialize() {
+        try {
+            setupDevice();
+            setupRefreshInterval();
+        } catch (ConfigurationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+    }
+
+    protected PJLinkDeviceConfiguration getConfiguration() throws ConfigurationException {
+      PJLinkDeviceConfiguration config = this.config;
+      try {
+          this.validateConfigurationParameters(this.getThing().getConfiguration().getProperties());
+      } catch(ConfigValidationException e) {
+          String message = e
+            .getValidationMessages().entrySet()
+            .stream()
+            .map((Map.Entry<String, String> a) -> (a.getKey() + ": " + a.getValue()))
+            .collect(Collectors.joining( "; "));
+          throw new ConfigurationException(message);
+      }
+      if(config == null) {
+        this.config = config = getConfigAs(PJLinkDeviceConfiguration.class);
+      }
+      return config;
+    }
+
+    private void clearRefreshInterval() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
```

Comment:
Just a remark, not a request for change.
I see you are using the same pattern several times in your code. You do not need to use a local variable. You could directly use the object member.

Reply:
This pattern had to be introduced when @NonNullByDefault was added.

First, the main reason why @NonNullByDefault makes sense are compiler warnings, so these have to be a clean to gain any benefit. Otherwise (if you just get accustomed to them), you won't notice if there are new ones you need to check.

So without the pattern you noticed, the compiler warns (correctly!) that the variable this.myvar could be null, even if I just checked that it's not null in the line before (due to potential concurrent modifications). The local variable ensures that it cannot have been nulled in the meantime -> no warning -> no hiding of potentially important warnings.

To summarize: @NonNullByDefault does not make sense if you do not also refactor your code to eliminate compiler warnings.

Reply:
Ok, you're right, I did not know the reason why the compiler was saying the var can be null.